tests/framework/pkg/session/session.go

```diff
@@ -31,15 +32,33 @@ func (ts *Session) RegisterCleanupFunc(f CleanupFunc) {
 
 }
 
+// RegisterWaitFunc
+func (ts *Session) RegisterWaitFunc(f CleanupFunc) {
```

Comment:
I don't think wait's should have their own queue, if a cleanup needs to wait before continuing it should block in it's cleanup function.  Cleanups should be sequential and the cleanup function should determine what is considered done before returning.

Reply:
I see what your saying, but when we are cleaning up we don't have the knowledge of the type of resource we are cleaning up in our cleanup func. Therefore we cannot get the specific watch.Interface for that resource.